### ISSUE TITLE
Crash when using d3d12

### ISSUE THREAD

**Issue by aGuyWhoMadeGames** [godotengine/godot#94148]
### Tested versions

- Reproducible in v4.3.beta2.official [b75f0485b]
- Not reproducible in v4.3.beta1.official [a4f2ea91a]

### System information

Godot v4.3.beta2 - Windows 10.0.19045 - d3d12 (Forward+) - dedicated NVIDIA GeForce GTX 960 (NVIDIA; 31.0.15.3623) - AMD Ryzen 5 1600 Six-Core Processor (12 Threads)

### Issue description

Running the following command opens a window that closes after a few seconds and prints the following errors.
`Godot_v4.3-beta2_win64.exe --rendering-driver d3d12 -v`

<details><summary>Output</summary>

```
Godot Engine v4.3.beta2.official.b75f0485b - https://godotengine.org
TextServer: Added interface "Dummy"
TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
Devices:
  #0: NVIDIA NVIDIA GeForce GTX 960 - Supported, Discrete
  #1: Microsoft Microsoft Basic Render Driver - Supported, CPU
- Shader:
  model: 6.5
- D3D12 Variable Rate Shading not supported
- D3D12 multiview supported:
  max view count: 4
- Relaxed casting not supported
- D3D12 16-bit ops supported: no
D3D12 12_0 - Forward+ - Using Device #0: NVIDIA - NVIDIA GeForce GTX 960
ERROR: Can't create buffer of size: 262144, error 0x80070057.
   at: (drivers/d3d12/rendering_device_driver_d3d12.cpp:917)
ERROR: Condition "!block.driver_id" is true. Returning: ERR_CANT_CREATE
   at: _insert_staging_block (servers/rendering/rendering_device.cpp:242)
ERROR: Condition "err != OK" is true. Continuing.
   at: initialize (servers/rendering/rendering_device.cpp:5510)
ERROR: Can't create buffer of size: 262144, error 0x80070057.
   at: (drivers/d3d12/rendering_device_driver_d3d12.cpp:917)
ERROR: Condition "!block.driver_id" is true. Returning: ERR_CANT_CREATE
   at: _insert_staging_block (servers/rendering/rendering_device.cpp:242)
ERROR: Condition "err != OK" is true. Continuing.
   at: initialize (servers/rendering/rendering_device.cpp:5510)
WARNING: PSO caching is not implemented yet in the Direct3D 12 driver.
     at: pipeline_cache_create (drivers/d3d12/rendering_device_driver_d3d12.cpp:4941)
Using "winink" pen tablet driver...
ERROR: FATAL: Index p_index = 0 is out of bounds (size() = 0).
   at: get (./core/templates/cowdata.h:205)

================================================================
CrashHandlerException: Program crashed with signal 4
Engine version: Godot Engine v4.3.beta2.official (b75f0485ba15951b87f1d9a2d8dd0fcd55e178e4)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] error(-1): no debug info in PE/COFF executable
[2] error(-1): no debug info in PE/COFF executable
[3] error(-1): no debug info in PE/COFF executable
[4] error(-1): no debug info in PE/COFF executable
[5] error(-1): no debug info in PE/COFF executable
[6] error(-1): no debug info in PE/COFF executable
[7] error(-1): no debug info in PE/COFF executable
[8] error(-1): no debug info in PE/COFF executable
[9] error(-1): no debug info in PE/COFF executable
[10] error(-1): no debug info in PE/COFF executable
[11] error(-1): no debug info in PE/COFF executable
[12] error(-1): no debug info in PE/COFF executable
[13] error(-1): no debug info in PE/COFF executable
[14] error(-1): no debug info in PE/COFF executable
[15] error(-1): no debug info in PE/COFF executable
[16] error(-1): no debug info in PE/COFF executable
[17] error(-1): no debug info in PE/COFF executable
[18] error(-1): no debug info in PE/COFF executable
[19] error(-1): no debug info in PE/COFF executable
[20] error(-1): no debug info in PE/COFF executable
-- END OF BACKTRACE --
================================================================


```
</details> 

The same thing happens when opening a project that uses d3d12. No error occurs in beta1 or dev6. I believe I have the correct dlls and folder structure.
```
│   D3D12Core.dll
│   d3d12SDKLayers.dll
│   dxil.dll
│   godot.windows.editor.x86_64.exe
│   Godot_v4.3-beta1_win64.exe
│   Godot_v4.3-beta2_win64.exe
│   Godot_v4.3-dev6_win64.exe
│
└───x86_64
        dxil.dll
```

### Steps to reproduce

 Run `Godot_v4.3-beta2_win64.exe --rendering-driver d3d12 -v`

### Minimal reproduction project (MRP)

N/a

**Comment by aGuyWhoMadeGames** [comment]
Running on v4.3.beta.gh-94131 [2b4861a20] (master as of now) does work, so it seems like the issue has either been solved, or the builds on https://godotengine.github.io/godot-commit-artifacts/ are somehow different from the builds on https://godotengine.org/download/archive/

**Comment by alvinhochun** [comment]
Duplicate of #93670, fixed by #93707

**Comment by Calinou** [comment]
Thanks for the report! Consolidating in https://github.com/godotengine/godot/issues/93670.

> or the builds on [godotengine.github.io/godot-commit-artifacts](https://godotengine.github.io/godot-commit-artifacts/) are somehow different from the builds on [godotengine.org/download/archive](https://godotengine.org/download/archive/)

CI builds reflect the state of the `master` branch, while dev/beta/RC builds are tagged on github.com/godotengine/godot-builds.
